src/dotnet/APIView/APIViewWeb/wwwroot/js/site.js

```diff
@@ -25,18 +25,60 @@ $(function () {
                 thread.after(commentForm);
             }
             else {
-                commentForm.insertAfter(thisRow).wrap("<tr>").wrap("<td>");
+                commentForm.insertAfter(thisRow).wrap("<tr>").wrap("<td colspan=\"2\">");
             }
         }
 
         commentForm.show();
         commentForm.find(".id-box").val(id);
         commentForm.find(".new-thread-comment-text").focus();
         commentForm.find(".comment-cancel-button").click(function () { hideCommentBox(id); });
+        commentForm.find(".comment-submit-button").click(function () {
+            $.ajax({
+                type: "POST",
+                data: commentForm.find("form").serialize()
+            }).done(function (partialViewResult) {
+                updateCommentThread(thisRow.next(), partialViewResult);
+            });
+            return false;
+        });
 
         nextRow.find(".review-thread-reply").hide();
     }
 
+    function updateCommentThread(commentBox, partialViewResult) {
+        partialViewResult = $.parseHTML(partialViewResult);
+        if ($(partialViewResult).find(".review-comment").length === 0) {
+            $(commentBox).remove();
```

Comment:
Why do you need to remove if you are replacing it with anyway?

Reply:
As the partial view currently renders, replacing the comment box with an empty one results in a thread that contains no comments but a reply button. I could alternately return an empty row for the partial view if there are no comments, but then replacing the thread with an empty row leaves a bit of empty space between comments that I don't think makes sense to leave in the HTML.

Reply:
You would have to show me this. I have hard time imagining :)